torax/sources/electron_cyclotron_source.py

```diff
@@ -0,0 +1,178 @@
+from __future__ import annotations
+
+import dataclasses
+
+import chex
+import jax
+import jax.numpy as jnp
+
+from torax import array_typing
+from torax import geometry
+from torax import interpolated_param
+from torax import state
+from torax.config import runtime_params_slice
+from torax.constants import CONSTANTS
+from torax.sources import runtime_params as runtime_params_lib
+from torax.sources import source
+
+
+@dataclasses.dataclass(kw_only=True)
+class RuntimeParams(runtime_params_lib.RuntimeParams):
+    """Runtime parameters for the electron-cyclotron source."""
+
+    # Global dimensionless current drive efficiency
+    global_efficiency: (
+        runtime_params_lib.interpolated_param.InterpolatedVarSingleAxisInput
+    )
+
+    # EC power density profile on the rho grid; units [W/m^2]
+    ec_power_density: runtime_params_lib.interpolated_param.InterpolatedVarTimeRhoInput
```

Comment:
is it OK if there is no default here? Maybe better to include zeros as default

Reply:
Or alternatively None as default. Depends on what choices are made vs Gaussian inputs

Reply:
Default now is zeros.